File: api-python-fastapi/database/cassandra.py
import os
from cassandra.cluster import Cluster
from cassandra.auth import PlainTextAuthProvider
from dotenv import load_dotenv

load_dotenv()

CASSANDRA_CONTACT_POINTS = os.getenv("CASSANDRA_CONTACT_POINTS", "127.0.0.1").split(",")
CASSANDRA_PORT = int(os.getenv("CASSANDRA_PORT", 9042))
CASSANDRA_KEYSPACE = os.getenv("CASSANDRA_KEYSPACE", "keyspace")
CASSANDRA_LOCAL_DC = os.getenv("CASSANDRA_LOCAL_DC", "datacenter")
CASSANDRA_USER = os.getenv("CASSANDRA_USER", "user")
CASSANDRA_PASSWORD = os.getenv("CASSANDRA_PASSWORD", "password")

auth_provider = None
if CASSANDRA_USER and CASSANDRA_PASSWORD:
    auth_provider = PlainTextAuthProvider(username=CASSANDRA_USER, password=CASSANDRA_PASSWORD)

cluster = Cluster(
    contact_points=CASSANDRA_CONTACT_POINTS,
    port=CASSANDRA_PORT,
    auth_provider=auth_provider,
)
session = cluster.connect()

# Ensure keyspace exists
session.execute(f"""
    CREATE KEYSPACE IF NOT EXISTS {CASSANDRA_KEYSPACE}
    WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': 1}};
""")
session.set_keyspace(CASSANDRA_KEYSPACE)

def get_cassandra_session():
    return session

File: api-nodejs-express/config/cassandra.js
const cassandra = require('cassandra-driver');
const { loadEnvFile } = require('node:process');
loadEnvFile();

const contactPoints = (process.env.CASSANDRA_CONTACT_POINTS || '127.0.0.1').split(',');
const port = parseInt(process.env.CASSANDRA_PORT, 10) || 9042;
const keyspace = process.env.CASSANDRA_KEYSPACE || 'keyspace';
const localDataCenter = process.env.CASSANDRA_LOCAL_DC || 'datacenter';

const authProvider = (process.env.CASSANDRA_USER && process.env.CASSANDRA_PASSWORD) 
    ? new cassandra.auth.PlainTextAuthProvider(process.env.CASSANDRA_USER, process.env.CASSANDRA_PASSWORD)
    : undefined;

const client = new cassandra.Client({
    contactPoints,
    localDataCenter,
    authProvider,
    protocolOptions: { port }
});

let initialized = false;

async function getSession() {
    if (!initialized) {
        await client.connect();
        
        await client.execute(`
            CREATE KEYSPACE IF NOT EXISTS ${keyspace}
            WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}
        `);

        await client.execute(`USE ${keyspace}`);

        await client.execute(`
            CREATE TABLE IF NOT EXISTS readings (
                device_id uuid,
                bucket_date text,
                ts bigint,
                sensor_values text,
                PRIMARY KEY ((device_id, bucket_date), ts)
            ) WITH CLUSTERING ORDER BY (ts DESC)
        `);

        initialized = true;
    }
    return client;
}

module.exports = { getSession };
